packages/vite-node/src/server.ts

```diff
@@ -412,7 +412,7 @@ export class ViteNodeServer {
     }
 
     const sourcemap = this.options.sourcemap ?? 'inline'
-    if (sourcemap === 'inline' && result && !id.includes('node_modules')) {
+    if (sourcemap === 'inline' && result && !(await this.shouldExternalize(id))) {
```

Comment:
This doesn't make much sense because it will alway return `true` - this function is not called otherwise:

https://github.com/vitest-dev/vitest/blob/51319929e5e25069ce57f15f8a4cdccb314b65ad/packages/vite-node/src/server.ts#L347-L358

We don't inline source map because it's expensive to do for every file in node modules

Reply:
Is there a way for me as a vitest user force it to generate source maps for specific modules from node_modules? In my case I'd like to generate source maps for the deps that I inline (`server.deps.inline`) because of #5605

Reply:
No, there is no way to do that, all `node_modules` files ignore stack trace. Maybe we can add another option 🤷🏻

Reply:
A new option in the config would solve my usecase 💯!

If it makes sense to allow users to generate source maps for not necessairly inlined deps then `server.deps.sourcemap` accepting `(string | RegExp)[]` seem ok, but if it makes sense only for inline deps then how about extending `server.deps.inline` like this
```diff
-inline: (string | RegExp)[] | true
+inline: (string | RegExp)[] | true | { match: string | RegExp, sourcemap: boolean }[]
```

Reply:
Ok thanks!
I'll close this PR then and see if I'll be able to implement the `includeSourcemap` option